app/routes/goal_routes.py

```diff
@@ -1 +1,88 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request
+from app.models.goal import Goal
+from app.models.task import Task
+from app.db import db
+from .route_utilities import validate_model, create_model, get_models_with_filters
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@goals_bp.post("")
+def create_goal(): 
+    request_body = request.get_json()
+    return create_model(Goal, request_body)
+
+@goals_bp.get("")
+def get_all_goals(): 
+    return get_models_with_filters(Goal, request.args)
+
+@goals_bp.get("/<goal_id>")
+def get_single_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    return {"goal":goal.to_dict()}
+
+@goals_bp.put("/<goal_id>")
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+    db.session.commit()
```

Comment:
Isn't this code similar to our other `.put` method? How could we use the `hasattr` and `setattr` functions that Python provides in order to make DRY up our routes?